src/amoc_distribution/amoc_cluster.erl

```diff
@@ -84,22 +117,37 @@ init(Nodes) ->
     {ok, NewState}.
 
 -spec handle_call(any(), any(), state()) -> {reply, any(), state()}.
-handle_call(get_master_node, _From, #state{master = Node} = State) ->
-    {reply, Node, State};
-handle_call({set_master_node, Node}, _From, #state{master = MasterNode} = State) ->
-    {NewState, RetValue} = case MasterNode of
+handle_call({set_master_node, Node}, _From, #state{master    = MasterNode,
+                                                   connected = Connected} = State) ->
+    {RetValue, NewState} = case MasterNode of
```

Comment:
I'm insisting on extracting this case into a dedicated function unless there is strong reason why it's not a good idea.